src/GameComponent/Game.jsx

```diff
@@ -0,0 +1,429 @@
+import React, { Component } from "react";
+import CountDownTimer from "../timerCounter/Countdown";
+import data from "../data/dictionary.json";
+import "./Game.css";
+class startScreen extends Component {
+  constructor(props) {
+    super(props);
+    this.currentState = {
+      name: "",
+      difficulty: 0,
+      selecteddifficulty: "Easy",
+      currDifficulty: 0,
+      startGameFlag: false,
+      startTimeFlag: false,
+      gameOverFlag: false,
+      currentWord: "",
+      currentScore: 0,
+      userInput: "",
+      easyWords: [],
+      mediumWords: [],
+      hardWords: [],
+      bestScore: 0,
+      bestScoreFlag: false,
+      bestScoreId: 0,
+      nameError: "",
+    };
+    this.state = { ...this.currentState };
+    this.scoreList = [];
+
+    //Spliting all the words into differenet variables
+    let easyWordFlag = true,
+      medWordFlag = true,
+      hardWordFlag = true;
+
+    for (let word of data) {
+      if (word.length <= 4 && easyWordFlag) {
+        this.state.easyWords.push(word);
+      } else if (word.length <= 8 && medWordFlag)
+        this.state.mediumWords.push(word);
+      else if (hardWordFlag) this.state.hardWords.push(word);
+
+      //For Optimization for delay
+      // if (this.state.easyWords.length === 150) easyWordFlag = false;
+      // if (this.state.mediumWords.length === 150) medWordFlag = false;
+      // if (this.state.hardWords.length === 500) hardWordFlag = false;
+    }
+  }
+  state = {};
+  //Method to save Name from UI
+  addName = (e) => {
+    this.setState({ name: e.target.value });
+  };
+  //Method to save selected Difficulty
+  selectDifficulty = (e) => {
+    this.setState({ selectDifficulty: e.target.value });
+  };
+  //Method to check whether Name has been written
+  validateName = () => {
+    if (this.state.name === "" || this.state.name.length === 0) {
```

Comment:
`this.state.name.length === 0` is sufficient in this case and covers first expression as well.